### ISSUE TITLE
Reminder keeps coming up after dismissing

### ISSUE THREAD

**Issue by sidz** [ExchangeCalendar/exchangecalendar#10]
I'm not sure that this fork will fix our issue but I want to try put that issue here as it exists in the forked repo: https://github.com/Ericsson/exchangecalendar/issues/546

Does anybody know how to fix it? or could anybody prepare a fix for that issue? I'm talking about normal fix not a uncheck 'Show missed reminders" option. 

Thanks mates. I hope this fork will prepare a fixes faster then previous.

**Comment by retorquere** [comment]
Same here. Can I just not have this popup show up at all? It takes some 5 minutes to keep clicking dismiss all when I just want to peek into my calendar.

**Comment by sidz** [comment]
@retorquere if you'll click by link which I provided you will find temporary workaround.

tldr: https://github.com/Ericsson/exchangecalendar/issues/546#issuecomment-266091072

**Comment by retorquere** [comment]
For those looking: I can't find that preference but in `about:config` I changed `calendar.alarms.showmissed` to `false` assuming it does the same.

**Comment by advancingu** [comment]
So the issue appears when you first receive a reminder for a repeating event that has already started before it was added to the user's own calendar? In other words, it will open a new reminder for every past event until the user has caught up dismissing to the current event?

**Comment by sidz** [comment]
@advancingu yes. But as for me it looks a little bit weird. Extension shouldn't show all events in the past.

**Comment by advancingu** [comment]
@Trim Do you know how "event notification seen" statuses are logged at the moment?

**Comment by Trim** [comment]
No, I don't know.

Today, I've checked a bit the git log and I've seen this commit: 1ab0995554ff0664dd360dc3f11714dcfc108b2f
It speaks about setting dismissing and enabling reminders.
It modifies the file `interfaces/exchangeBaseItem/mivExchangeBaseItem.js`.

I've tried to read this code, but it's very huge (around 4'000 lines of code!) and I don't really understand it. There's a ton of setters and getters, there was a lot of `dump` commented...

That's really a mess in my point of view and it will be difficult to improve it in short time: despite its name "BaseItem" there's a lot of things managed by it (attendees, alarms, task progression, attachments, ...).

In short time, I want be able to modify this code, it will ask me a lot of time and I'm not sure if I'll have enough time in the future to modify this code :'(

**Comment by logological** [comment]
I suffered from this problem for a long time.  Since upgrading to 4.0.0-beta3 from this project it no longer occurs.
